lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2385,7 +2385,31 @@ postTransactionFeeOld ctx@ApiLayer{..} (ApiT walletId) body = do
                     ^? #metadata . traverse . #txMetadataWithSchema_metadata
                 }
             PreSelection{outputs}
-        pure $ mkApiFee Nothing minCoins feePercentiles
+        pure
+            $ mkApiFee Nothing minCoins
+            $ W.padFeePercentiles protocolParameters padding feePercentiles
+  where
+    -- Padding to make the fee percentiles more imprecise, for the following
+    -- reasons:
+    --
+    -- 1. dummyChangeAddressGen uses the longest possbile addresses. For byron
+    -- wallets they are 83 bytes long, which is longer than what we'd
+    -- expect in reality, and longer than the shortest Byron address we'd expect
+    -- of 66 bytes. I.e. we could be off by up to 17 bytes.
+    --
+    -- 2. Our integration are making assumptions that the fees of transactions
+    -- alyways lie within the estimated interval. These are faulty - the
+    -- estimation is a pair of 10- and 90-th percentiles, not any absolute
+    -- bounds. But they are also useful for the moment. During the risky
+    -- transition to balanceTx they will temporarily compare a rewritten
+    -- postTransactionFee with an old postTransaction. They may also act as
+    -- golden tests in lieu of better/more ones on the unit level. A small
+    -- amount of padding is a cheap way to keep these checks.
+    --
+    -- In the context of a mainnet `minfeeA` value of 44 lovelace/byte the
+    -- padding is negligible - less than 1/1000 ada.
+    padding :: Quantity "byte" Word
+    padding = Quantity 20
```

Comment:
I found it difficult to separate this into multiple parts; I suspect the 3 in 17+3 would be too small.